### PR TITLE
feat(component): convert Tooltip to FC and remove static members

### PR DESCRIPTION
**Pull request by jordan-massingill**
### WHAT

Convert the Tooltip class component to a FC and remove its static members

### CODE REVIEW

File: packages/big-design/src/components/Tooltip/Tooltip.tsx [hunk 1]
```diff
@@ -13,90 +14,80 @@ export interface TooltipProps extends React.HTMLAttributes<HTMLDivElement> {
...
+        if (tooltipContainer) {
+          document.body.removeChild(tooltipContainer);
+        }
+      };
+    }, [tooltipContainer]);
 
-  private tooltipContainer?: HTMLDivElement;
+    const renderContent = (content: React.ReactNode) => {
+      return typeof content === 'string' ? <Small color="white">{content}</Small> : content;
+    };
```
Comment (chanceaclark) [inline packages/big-design/src/components/Tooltip/Tooltip.tsx]: 🤔 May want to refactor to something like this so the content doesn't cause any unnecessary rerenders.
```tsx
    const renderedContent = useMemo(() => {
      return typeof children === 'string' ? <Small color="white">{children}</Small> : children;
    }, [children]);

    ...further down...
                    <StyledTooltip ref={ref} style={style} data-placement={placement}>
                      {renderedContent}
                    </StyledTooltip>

```

File: packages/big-design/src/components/Tooltip/Tooltip.tsx [hunk 2]
```diff
@@ -13,90 +14,80 @@ export interface TooltipProps extends React.HTMLAttributes<HTMLDivElement> {
...
+    useEffect(() => {
+      return () => {
+        if (tooltipContainer) {
+          document.body.removeChild(tooltipContainer);
+        }
+      };
+    }, [tooltipContainer]);
 
-  private tooltipContainer?: HTMLDivElement;
+    const renderContent = useMemo(() => {
```
Comment (deini) [inline packages/big-design/src/components/Tooltip/Tooltip.tsx]: @jordan-massingill @chanceaclark I don't think this `useMemo` is needed here. `{renderContent}` is already  behind the `isVisible` check. When `isVisible = true` it has to re-render anyway.

File: packages/big-design/src/components/Tooltip/Tooltip.tsx [hunk 3]
```diff
@@ -13,90 +14,80 @@ export interface TooltipProps extends React.HTMLAttributes<HTMLDivElement> {
...
         </Reference>
-        {this.tooltipContainer
+        {tooltipContainer
           ? createPortal(
               <Popper
-                placement={this.props.placement}
-                modifiers={{ offset: { offset: '0, 8' }, ...this.props.modifiers }}
-                eventsEnabled={this.state.visible}
+                placement={props.placement || 'top'}
+                modifiers={{ offset: { offset: '0, 8' }, ...modifiers }}
```
Comment (jorgemoya) [inline packages/big-design/src/components/Tooltip/Tooltip.tsx]: I had an issue with this `offset` object. React wanted me to memoize it. If you're having no issues with this I need to look at my code haha.

File: packages/big-design/src/components/Tooltip/Tooltip.tsx [hunk 4]
```diff
@@ -13,90 +14,80 @@ export interface TooltipProps extends React.HTMLAttributes<HTMLDivElement> {
...
-  state = {
-    visible: false,
-  };
+    useEffect(() => {
+      return () => {
+        if (tooltipContainer) {
+          document.body.removeChild(tooltipContainer);
+        }
+      };
+    }, [tooltipContainer]);
```
Comment (jorgemoya) [inline packages/big-design/src/components/Tooltip/Tooltip.tsx]: I don't understand the purpose of this. You set the tooltipContainer up top but then remove it?

Comment (jordan-massingill) [reply packages/big-design/src/components/Tooltip/Tooltip.tsx]: Yes - this useEffect just handles the unmount.

File: packages/big-design/src/components/Tooltip/Tooltip.tsx (hunk 3 above)
Comment (jordan-massingill) [reply packages/big-design/src/components/Tooltip/Tooltip.tsx]: This hasn't given me any issues *_so far_*.

File: packages/big-design/src/components/Tooltip/Tooltip.tsx [hunk 5]
```diff
@@ -13,90 +13,78 @@ export interface TooltipProps extends React.HTMLAttributes<HTMLDivElement> {
...
+  useEffect(() => {
+    return () => {
+      if (tooltipContainer) {
+        document.body.removeChild(tooltipContainer);
+      }
+    };
+  }, [tooltipContainer]);
 
-  private renderContent = (content: React.ReactNode) => {
+  const renderContent = (content: React.ReactNode) => {
```
Comment (deini) [inline packages/big-design/src/components/Tooltip/Tooltip.tsx]: 🍹 Since you have access to `children` here I would use it directly instead of passing it.

Comment (jordan-massingill) [reply packages/big-design/src/components/Tooltip/Tooltip.tsx]: Yeah, I went back and forth on whether to pass it or use it directly. This implementation was only to cover the edge case that this function could be used elsewhere in future refactors (though unlikely). Will change it back. 👍

**Pull request merged**